docs/blog/2018-12-04-per-link-gatsby-page-transitions-with-transitionlink/index.md

```diff
@@ -0,0 +1,153 @@
+  <TransitionLink to={to} exit={{ length: 0.5 }}>
+    {children}
+  </TransitionLink>
+)
+```
+
+```jsx
+<FadeLink to="/page-2">Go to page 2</FadeLink>
+```
+
+The beauty of this is that if you decide a certain page requires something more complex than a fade, you can add more transitions as needed.
+
+Here's an example of a more complicated TransitionLink using [gsap](https://greensock.com) to create a coloured ripple that moves outwards from the users mouse click, eventually covering the whole page.
+
+```jsx
+<TransitionLink
+  exit={{
+    length: 0.6,
+    trigger: ({ exit, e, node }) =>
+      this.createRipple(exit, e, props.color, node),
+  }}
+  entry={{
+    delay: 0.3,
+    length: 0.6,
+    trigger: ({ entry, node }) => this.slideIn(entry, node, 'left'),
+  }}
+>
+  {props.children}
+</TransitionLink>
+```
+
+The animation code for `this.createRipple` would be a bit much to copy here but you can check it out on the [TransitionLink github](https://github.com/TylerBarnes/gatsby-plugin-transition-link/blob/master/src/AniLink/PaintDrip.js). If you'd like you can also try it out with AniLink's paintDrip transition (check the [AniLink docs](https://transitionlink.tylerbarnes.ca/docs/anilink/) for usage).
+
+As you can see, TransitionLink offers quite a wide variety of control for page transitions! You're able to set the length, delay, state, and a trigger function for both entering and exiting pages, allowing you to use both declarative and imperative animations and as many transitions as you need.
+
+### Optional components to help you out
+
+In addition to TransitionLink and AniLink, there's also the TransitionState and TransitionPortal components. While I was developing and testing TransitionLink it became obvious that these two components were needed to keep things manageable and get around some limitations.
+
+TransitionState uses children as a function to pass the transition status of the page the component is used on. That means you can use it throughout your pages to make your components aware of the transition status of the page it's a child of. You can also use this component to pass any kind of state from TransitionLink to your components.
+
+TransitionPortal is a React portal which allows you to position animation elements above both your exiting and entering pages. Sometimes you'll need to have something persist throughout the whole animation and some other page element will overlap it. Portals are great for solving that.
+
+[Check the docs](https://transitionlink.tylerbarnes.ca/docs) for more info on TransitionState and TransitionPortal.
+
+### Capping it off
+
+I'm excited to see what you and the rest of the Gatsby community will build with TransitionLink! Feel free to send any unanswered questions or issues you may have [my way](https://github.com/TylerBarnes/gatsby-plugin-transition-link/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc). I'll do my best to help you out asap.
+
+— Ty. B.
```

Comment:
We tend to avoid signatures on blog posts for consistency.

Reply:
That makes sense. I removed it